imagetool/src/main/java/com/oracle/weblogic/imagetool/util/HttpUtil.java

```diff
@@ -120,12 +124,20 @@ public static HttpClient getOraClient(String userId, String password) {
         String proxyHost = System.getProperty("https.proxyHost");
         String proxyPort  = System.getProperty("https.proxyPort");
         HttpClient result;
+        CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+
+        if (userId != null && password != null) {
+            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(
+                userId, password));
+        }
+
         result = HttpClientBuilder.create()
             .setDefaultRequestConfig(config.build())
             .setRetryHandler(retryHandler())
             .setProxy(proxyHost != null ? new HttpHost(proxyHost, Integer.parseInt(proxyPort)) : null)
             .setUserAgent("Wget/1.10")
             .setDefaultCookieStore(cookieStore).useSystemProperties()
+            .setDefaultCredentialsProvider(credentialsProvider)
```

Comment:
Why bother adding the credential provider when the userId and password are not supplied?  I would try to avoid this unless it is strictly required and only create the CredentialsProvider and add it to the request when the username and password are specified...

Reply:
I changed to create CredentialProvider only when username and password is provided.  It is required to authenticate against the support system to get patches information.